app/scripts/analysis/tools/powercalc/PowercalcMainCtrl.ctrl.coffee

```diff
@@ -0,0 +1,1529 @@
+'use strict'
+
+BaseCtrl = require 'scripts/BaseClasses/BaseController.coffee'
+
+module.exports = class PowercalcMainCtrl extends BaseCtrl
+  @inject 'app_analysis_powercalc_msgService',
+  'app_analysis_powercalc_TwoTGUI',
+  'app_analysis_powercalc_OneTGUI',
+  '$timeout',
+  '$scope'
+
+  initialize: ->
+    console.log("mainArea initialized")
+    @powerAnalysis = require 'powercalc'
+    @distribution = require 'distributome'
+    # require('mathjax')
+    # require('mathjax/unpacked/jax/input/TeX/config.js')
+    # require('mathjax/unpacked/jax/output/HTML-CSS/config.js')
+    # require('mathjax/unpacked/extensions/tex2jax.js')
+    # require('mathjax/unpacked/extensions/MathMenu.js')
+    # require('mathjax/unpacked/extensions/MathZoom.js')
+    # require('mathjax/unpacked/jax/element/mml/jax.js')
+    # require('mathjax/unpacked/jax/input/TeX/jax.js')
+    # require('mathjax/unpacked/jax/output/HTML-CSS/jax.js')
+    # require('mathjax/unpacked/extensions/MathEvents.js')
+    # require('mathjax/unpacked/jax/output/HTML-CSS/fonts/TeX/fontdata.js')
+    # require('mathjax/unpacked/jax/output/HTML-CSS/jax.js')
+    # require('mathjax/unpacked/jax/output/HTML-CSS/fonts/STIX/fontdata.js')
+    # require('mathjax/')
+    # require('mathjax/')
+    # require('mathjax/')
+    # require('mathjax/')
+    @msgService = @app_analysis_powercalc_msgService
+    @title = 'Power Calculator Module'
+    #algorithm type
+    @selectedAlgorithm = "Two-sample t test (general case)"
+    @SIGNIFICANT = 5
+    @data = []
+    @dataType = ''
+    @dataPoints = null
+    @means = null
+    @assignments = null
+    @populations = {}
+    @deployed = false
+    @chosenCols = []
+    @comp_agents = []
+    @agent = ""
+
+    #variables needed for cfap only
```

Comment:
defining parameters for all calculators here creates a mess.
parameters specific for each calculator should be separated into a individual file(s). as one example, see it's done for [cluster algorithms](https://github.com/SOCR/SOCRAT/blob/dev/app/scripts/analysis/tools/Cluster/ClusterAlgorithms.service.coffee#L5), where each algorithm is defined as a [separate service](https://github.com/SOCR/SOCRAT/blob/dev/app/scripts/analysis/tools/Cluster/ClusterAlgorithms.service.coffee#L7) and injected as a dependency and provides [`getParams` method](https://github.com/SOCR/SOCRAT/blob/dev/app/scripts/analysis/tools/Cluster/ClusterKMeans.service.coffee#L47), that allows [to get it's parameters by it's name](https://github.com/SOCR/SOCRAT/blob/dev/app/scripts/analysis/tools/Cluster/ClusterAlgorithms.service.coffee#L22)